WidgetKit/Helpers.swift

```diff
@@ -13,8 +14,30 @@ var scheme: String {
     return string
 }
 
-func linkToContainingApp(_ urlSuffix: String = "", query: String) -> URL {
+func linkToContainingApp(
+    _ urlSuffix: String = "",
+    query: String,
+    widgetObjectIdentifier: String,
+    widgetSize: WidgetSize
+) -> URL {
     let urlString = "\(scheme)://\(query)\(urlSuffix)"
+    
+    let prefs = BrowserProfile(localName: "profile").prefs
+    let prefKeySyncEvents = "widgetkit.telemetry.event"
+    var events = prefs.arrayForKey(prefKeySyncEvents) as? [Data] ?? []
```

Comment:
`if events.count > 20 { log to sentry that we are piling up too many events and return url }`